pkg/v4/resource/chartconfig/update.go

```diff
@@ -107,3 +100,43 @@ func (r *Resource) newUpdateChange(ctx context.Context, obj, currentState, desir
 
 	return chartConfigsToUpdate, nil
 }
+
+func (r *Resource) updateChartConfig(ctx context.Context, updatedChartConfig *v1alpha1.ChartConfig, guestG8sClient versioned.Interface) error {
+	chartConfigName := updatedChartConfig.ObjectMeta.Name
+
+	chartConfig, err := guestG8sClient.CoreV1alpha1().ChartConfigs(resourceNamespace).Get(chartConfigName, metav1.GetOptions{})
```

Comment:
Why don't we use the information we get from `GetCurrentState`? When we compute the update patch we could just leverage all information we have available and prevent the additional call here.

Reply:
Good point. I'll try that. It would be good to avoid the extra API call.